app/assets/javascripts/controllers/ems_common/ems_common_form_controller.js

```diff
@@ -318,6 +326,18 @@ ManageIQ.angular.app.controller('emsCommonFormController', ['$http', '$scope', '
     }
   };
 
+  $scope.detectClicked = function($event) {
+    miqService.sparkleOn();
+    miqService.detectWithRest($event, $scope.actionUrl)
+      .then(function success(data) {
```

Comment:
This callback also handles detection failure as an error is returned (tested) similarly to the [validate function](https://github.com/manageiq/manageiq-ui-classic/blob/197079ad373bce1d8c8088d951859096dcb7e5db/app/assets/javascripts/controllers/ems_common/ems_common_form_controller.js#L517)

as for failure to get the response there's not much we can do if we can't talk to the server - so I think it is good enough to do the same as the validate callback.